### PR TITLE
Show segments on contact detail

### PR DESCRIPTION
**Pull request by snoblucha**
| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | 3.3
| Bug fix?                               | no
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | 
| Related developer documentation PR URL |
| Issue(s) addressed                     | 

just a little enhancment of contact details. It adds leads segments under the companies list with links to the segments.

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10160"></a>

### CODE REVIEW

**Comment by npracht** [comment]
Hi @snoblucha !
Thanks a lot for this nice contribution. Could you please rebase your PR against `features` branch ?

File: plugins/MauticCrmBundle/Integration/CrmAbstractIntegration.php [hunk 1]
```diff
@@ -429,7 +429,7 @@ public function getMauticLead($data, $persist = true, $socialCache = null, $iden
             }
         }
 
-        $leadModel->setFieldValues($lead, $matchedFields, false, false);
+        $leadModel->setFieldValues($lead, $matchedFields, true, false);
```
Comment (kuzmany) [inline plugins/MauticCrmBundle/Integration/CrmAbstractIntegration.php]: This is not related to this PR, right?

**Review by kuzmany — DISMISSED** [review]
This is nice addition and works well. Just need remove commented code as not part of this PR 👍

File: plugins/MauticCrmBundle/Integration/CrmAbstractIntegration.php (hunk 1 above)
Comment (snoblucha) [reply plugins/MauticCrmBundle/Integration/CrmAbstractIntegration.php]: No, this was commited by mistake.

File: app/bundles/LeadBundle/Views/Lead/lead.html.php [hunk 2]
```diff
@@ -597,6 +597,21 @@ class="fa fa-check <?php echo (1 == $company['is_primary']) ? 'primary' : ''; ?>
...
             <div class="clearfix"></div>
         </div>
+        <div class="pa-sm panel-segments">
+          <div class="panel-title">
+            <?php echo $view['translator']->trans('mautic.lead.lead.lists'); ?>
+          </div>
+          <?php foreach ($lists[$lead->getId()] as $key => $list): ?>
+            <h5 class="pull-left mt-xs mr-xs">
+                  <span class="label label-success">
+                      <a href="<?php echo $view['router']->path('mautic_segment_action', ['objectAction' => 'view', 'objectId' => $list['id']]); ?>" style="color: white;">
```
Comment (kuzmany) [inline app/bundles/LeadBundle/Views/Lead/lead.html.php]: If you add data attribute  `data-toggle="ajax"` to anchor, links will loaded with ajax (same in companies).
Can you do that?

File: plugins/MauticCrmBundle/Integration/CrmAbstractIntegration.php (hunk 1 above)
Comment (snoblucha) [reply plugins/MauticCrmBundle/Integration/CrmAbstractIntegration.php]: removed

File: app/bundles/LeadBundle/Views/Lead/lead.html.php (hunk 2 above)
Comment (snoblucha) [reply app/bundles/LeadBundle/Views/Lead/lead.html.php]: Great point. I've added it to both links.

**Review by kuzmany — DISMISSED** [review]
Looks good 👍 
Nice feature
Approve it

**Comment by RCheesley** [comment]
@snoblucha it looks like there is a failing test to be addressed, would you be able to take a look?

We're hoping to get this merged with the 4.2 release so we have a bit of time!

**Comment by kuzmany** [comment]
@RCheesley why 4.2? Why not 4.1? Does not make sense

**Comment by RCheesley** [comment]
@kuzmany here is the release planning process we are now following since the 4.0 release: https://www.mautic.org/blog/community/updates-mautic-community-release-process.

We had the planning meeting for the 4.1.1, 4.1.2 and 4.2 release yesterday where we reviewed open PRs on the themes we are covering for the releases (Segments, Contacts and Companies) and considered which ones we might think about targeting to get addressed. You can see the release planning spreadsheet here: https://docs.google.com/spreadsheets/d/11DF0o3guxr5yGYAqoWC9JFJkYU35NPRu3nQOreiUAyI/edit?usp=sharing.  Everybody is welcome to join these meetings, the dates are advertised in the blog post I mentioned above and on the community calendar, and the link is shared in the Product Team Slack channel.

We are finding that this approach is really helping the release leaders to focus their attention on specific PRs, instead of having to sift through all 200+ PRs and figure out which ones to try to get over the line.  It also helps the marketers when we have a bunch of fixes in specific areas for a release (plus any others which happen to be ready to commit at the time of the release) and also means we can be more thorough with testing if we know most of the fixes are in a certain area or feature.

Past experience has shown us that we have to be sensible about what we are trying to achieve with each release, so we are doing our best to target specific PRs that we 'definitely' want to get into each release during the planning meeting, which are added to the milestone **when they are in a state that means they can be merged** - e.g. tests passing and any outstanding feedback being addressed fully. We found several PRs, like this one, which we want to consider for the release but which are not in a state where they could be merged and need further work. Hence the message asking if the author can fix the issues so we can consider it for the milestone.

If we can get the PR fixed and tested/code reviewed sooner than 4.2 then great, it can go into 4.1. But the absolute worst case, we'd like to get it into 4.2. Hopefully this also helps developers with a large number of outstanding PRs with issues to prioritise the ones that are hoping to get into an upcoming release, and helps testers to get clarity on where we really need their involvement.

I hope that explains why we are taking this approach - it is all outlined in the blog post above, feel free to discuss in the Product Team Slack channel if anything is unclear!

**Review by kuzmany — DISMISSED** [review]

**Review by npracht — DISMISSED** [review]
👍🏽 works well, nice UI improvement (today you need to click on preferences, etc. to see contact segments).
Thanks !

We're good to go in next minor release :)

**Review by dennisameling — APPROVED** [review]
